### DISCUSSION TITLE
Welcome to foundry-vtt-tidy-5e-sheets Discussions!

### DISCUSSION THREAD

**Discussion by kgar** [kgar/foundry-vtt-tidy-5e-sheets#89] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by WThunderspirit** [comment]
I can't upgrade beyond D&D 2.3 because of me and my group's hardware and local internet limitations.  Is there any way to recover the previously compatible version for me or is it gone forever as technology goes beyond my computer and local provider settings?  The option is still there for me to download, but doesn't appear in game at all because Foundry is probably detecting incompatibility.  I was wondering if I could do a hard install of the previous files without autoupdating to maintain what I have had.

**Reply by kgar** [reply]
Yes, I maintain a compatibility table on the official package page: https://foundryvtt.com/packages/tidy5e-sheet/

There is a Tidy version that is compatible with dnd5e 2.2.x-2.3.x:
![image](https://github.com/kgar/foundry-vtt-tidy-5e-sheets/assets/4632129/f9bc6c9f-d237-4c8a-a971-63b2c565c0c6)

The compatible manifest: https://github.com/sdenec/tidy5e-sheet/releases/download/0.9.19/module.json

Perform a URL-based installation of the target module.json link, and then lock the package from updating. That should allow you to continue using Tidy5e Sheet.

**Comment by Kir6in** [comment]
Is it possible to get a non-legacy version of the Tidy sheet blank, but a regular one on the 12.311 version of Foundry? I want to make different sheet colors for each player, and this is the most possible way with your module

**Reply by kgar** [reply]
If you are asking about bringing the new Tidy sheet design to previous versions of Foundry/dnd5e, it is far too much work to port the new Tidy sheets back to Foundry V12 / dnd5e 4.x or earlier. I am not willing to take on this project.

**Reply by Kir6in** [reply]
Can you recommend a way to set a different color theme for each player's sheet?

**Reply by kgar** [reply]
There is one way with the classic sheets.
Using the Custom CSS module, you can add a block of CSS for each actor to fully customize their colors.

```css
/* 
    For each actor, create a block of CSS like this, and paste the actor's UUID to overwrite "DOCUMENT-UUID-HERE"; 
    for example, my Akra actor would be  
    .tidy5e-sheet.classic[data-document-uuid="Actor.cayb9oZcp9JFXfLf"] { 
*/
.tidy5e-sheet.classic[data-document-uuid="DOCUMENT-UUID-HERE"] {
  --t5e-primary-accent-color: rebeccapurple !important;
  --t5e-hp-bar-color: rebeccapurple !important;
  --t5e-equipped-background: rebeccapurple !important;
  --t5e-equipped-item-grid-tile-outline-color: rebeccapurple !important;
  --t5e-equipped-item-grid-tile-accent-color: rebeccapurple !important;
  --t5e-prepared-background: rebeccapurple !important;
  --t5e-prepared-item-grid-tile-outline-color: rebeccapurple !important;
  --t5e-prepared-item-grid-tile-accent-color: rebeccapurple !important;
  --t5e-pact-background: rebeccapurple !important;
  --t5e-pact-outline-color: rebeccapurple !important;
  --t5e-pact-accent-color: rebeccapurple !important;
  --t5e-atwill-background: rebeccapurple !important;
  --t5e-atwill-outline-color: rebeccapurple !important;
  --t5e-atwill-accent-color: rebeccapurple !important;
  --t5e-ritual-only-background: rebeccapurple !important;
  --t5e-ritual-only-outline-color: rebeccapurple !important;
  --t5e-ritual-only-accent-color: rebeccapurple !important;
  --t5e-innate-background: rebeccapurple !important;
  --t5e-innate-outline: rebeccapurple !important;
  --t5e-innate-accent: rebeccapurple !important;
  --t5e-alwaysprepared-background: rebeccapurple !important;
  --t5e-alwaysprepared-outline-color: rebeccapurple !important;
  --t5e-alwaysprepared-accent-color: rebeccapurple !important;
  --t5e-scrollbar-thumb-color: rebeccapurple !important;
  --t5e-scrollbar-track-color: rebeccapurple !important;
  --t5e-magic-accent-color: rebeccapurple !important;
}

```

I have added `rebeccapurple` as my sample color, but you can fill in any color for these values.
You can also delete any of the colors you do not wish to customize.

To copy the UUID, click on the header control:

<img width="259" height="205" alt="image" src="https://github.com/user-attachments/assets/554f6120-878a-4ecd-bf06-2e1fe266e5fb" />

Before applying custom CSS to my Akra test actor:

<img width="850" height="893" alt="image" src="https://github.com/user-attachments/assets/76f00fd2-638d-4676-add2-e66ed5485161" />

After applying this CSS

```css
.tidy5e-sheet.classic[data-document-uuid="Actor.cayb9oZcp9JFXfLf"] {
  --t5e-equipped-background: rebeccapurple !important;
  --t5e-equipped-item-grid-tile-outline-color: rebeccapurple !important;
  --t5e-equipped-item-grid-tile-accent-color: rebeccapurple !important;
  --t5e-prepared-background: rebeccapurple !important;
  --t5e-prepared-item-grid-tile-outline-color: rebeccapurple !important;
  --t5e-prepared-item-grid-tile-accent-color: rebeccapurple !important;
  --t5e-pact-background: rebeccapurple !important;
  --t5e-pact-outline-color: rebeccapurple !important;
  --t5e-pact-accent-color: rebeccapurple !important;
  --t5e-atwill-background: rebeccapurple !important;
  --t5e-atwill-outline-color: rebeccapurple !important;
  --t5e-atwill-accent-color: rebeccapurple !important;
  --t5e-ritual-only-background: rebeccapurple !important;
  --t5e-ritual-only-outline-color: rebeccapurple !important;
  --t5e-ritual-only-accent-color: rebeccapurple !important;
  --t5e-innate-background: rebeccapurple !important;
  --t5e-innate-outline: rebeccapurple !important;
  --t5e-innate-accent: rebeccapurple !important;
  --t5e-alwaysprepared-background: rebeccapurple !important;
  --t5e-alwaysprepared-outline-color: rebeccapurple !important;
  --t5e-alwaysprepared-accent-color: rebeccapurple !important;
  --t5e-scrollbar-thumb-color: rebeccapurple !important;
  --t5e-scrollbar-track-color: rebeccapurple !important;
  --t5e-magic-accent-color: rebeccapurple !important;
}
```

<img width="845" height="878" alt="image" src="https://github.com/user-attachments/assets/f248a3bb-6acc-4d95-9cd1-661a195f40f0" />

---

Give this a try and see how it works out.

**Comment by Kir6in** [comment]
I tried something like that to Default 5e character sheet without Tidy Sheets module
<img width="1290" height="899" alt="image" src="https://github.com/user-attachments/assets/5a125c52-1f09-4f49-923f-05868b866d58" />
But one of the problems was adjusting the style of the container sheet to the character's theme. Every time the container is removed from the inventory, its name changes, and I have to reassign it manually to make the style display correctly again. I tried to push the css styles by getting a list of all the character's container styles, but it was too complicated.

**Reply by kgar** [reply]
With Tidy and Default sheets, you can target the ID using a wildcard selector to indicate "Select all item sheets belonging to a specific actor":

```css
[id*="Actor-tkNVBYeoPy8zCax1-Item-"] {
    background: pink !important;
}
```

Explanation: This selector targets all items belonging to actor `Actor.tkNVBYeoPy8zCax1`, because their UUIDs will generally be `Actor.tkNVBYeoPy8zCax1.Item.SomeItemIDHere`. However, IDs cannot have dots, so replace dots with hyphens "-", which is what the Foundry application does. The selector `[id*=""]` means "IDs which contains this text".

Note that you can also just generically target ALL sheets belonging to your actor to apply universal styles:

```css
[id*="Actor-tkNVBYeoPy8zCax1"] {
    background: pink !important;
}

```

**Reply by Kir6in** [reply]
I did it! It works properly now.

```
:root {
--P1-main: rgb(83 63 103);
--P1-sec: rgb(57 42 70);
--P1-filters: grayscale(15%) brightness(120%) hue-rotate(278deg) saturate(50%) contrast(100%);
}

.sheet[id*="Actor-VBkFX5CgiIg6qQGT"] .items-header{
background: linear-gradient(to right, var(--P1-main), var(--P1-sec)) !important;
}

.sheet[id*="Actor-VBkFX5CgiIg6qQGT"] section.window-content::before{
filter: var(--P1-filters);
}
```
Thank you!
